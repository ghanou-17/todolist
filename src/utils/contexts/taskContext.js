import { createContext } from "react";
// Cree le contexte, il est ici uniquemement cree, il n'est pas encore utilise
// Il sera utilise dans src/components/todoList/index.js
// C'est pour faire propre
export const TasksContext = createContext();
