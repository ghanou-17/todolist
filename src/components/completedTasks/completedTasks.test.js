import React from "react";
import { render, screen } from "@testing-library/react";
import { TasksContext } from "../../utils/contexts/taskContext";
import CompletedTasks from "./index";

describe("CompletedTasks", () => {
  const renderWithTasksContext = (taskList) => {
    return render(
      <TasksContext.Provider value={{ tasks: taskList }}>
        <CompletedTasks />
      </TasksContext.Provider>,
    );
  };

  test("it should render no tasks when there are no completed tasks", () => {
    renderWithTasksContext([{ id: 1, text: "Task 1", completed: false }]);
    expect(screen.queryByText("Task 1")).toBeNull();
  });

  test("it should render completed tasks", () => {
    const tasks = [
      { id: 1, text: "Task 1", completed: false },
      { id: 2, text: "Task 2", completed: true },
      { id: 3, text: "Task 3", completed: true },
    ];
    renderWithTasksContext(tasks);
    expect(screen.getByText("Task 2")).not.toBeNull();
    expect(screen.getByText("Task 3")).not.toBeNull();
  });

  test("it should not render incomplete tasks", () => {
    const tasks = [
      { id: 1, text: "Task 1", completed: false },
      { id: 2, text: "Task 2", completed: true },
    ];
    renderWithTasksContext(tasks);
    expect(screen.queryByText("Task 1")).toBeNull();
  });

  test("it should match snapshot with completed tasks", () => {
    const tasks = [
      { id: 1, text: "Task 1", completed: true },
      { id: 2, text: "Task 2", completed: true },
    ];
    const { asFragment } = renderWithTasksContext(tasks);
    expect(asFragment()).toMatchSnapshot();
  });
});
