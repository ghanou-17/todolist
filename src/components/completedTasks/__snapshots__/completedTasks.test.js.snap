// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedTasks it should match snapshot with completed tasks 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Tâches terminées
    </h2>
    <ul>
      <li>
        Task 1
      </li>
      <li>
        Task 2
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
